checkstyle.xml

```diff
@@ -0,0 +1,122 @@
+-->
+<module name="Checker">
+    <!-- Configure checker to use UTF-8 encoding -->
+    <property name="charset" value="UTF-8"/>
+    <!-- Configure checker to run on files with these extensions -->
+    <property name="fileExtensions" value="java, properties, cfg, xml"/>
+
+    <!-- No tab characters ('\t') allowed in the source code -->
+    <module name="FileTabCharacter">
+        <property name="eachLine" value="true"/>
+        <property name="fileExtensions" value="java, properties, cfg, css, js, xml"/>
+    </module>
+
+    <!-- No Trailing Whitespace, except on lines that only have an asterisk (e.g. Javadoc comments) -->
+    <module name="RegexpSingleline">
+        <property name="format" value="(?&lt;!\*)\s+$|\*\s\s+$"/>
+        <property name="message" value="Line has trailing whitespace"/>
+        <property name="fileExtensions" value="java, properties, cfg, css, js, xml"/>
+    </module>
+
+    <!-- Allow individual lines of code to be excluded from these rules, if they are annotated
+         with @SuppressWarnings. See also SuppressWarningsHolder below -->
+    <module name="SuppressWarningsFilter" />
+
+    <!-- Check individual Java source files for specific rules -->
+    <module name="TreeWalker">
+        <!-- Maximum line length is 120 characters -->
+        <module name="LineLength">
+            <property name="max" value="120"/>
+        </module>
+
+        <!-- Check for and report back any TODO or FIXME comments as info messages -->
+        <module name="TodoComment">
+            <property name="severity" value="info"/>
+            <property name="format" value="(TODO)|(FIXME)"/>
+        </module>
+
+        <!-- Do not report errors on any lines annotated with @SuppressWarnings -->
+        <module name="SuppressWarningsHolder"/>
+
+        <!-- ##### Import statement requirements ##### -->
+        <!-- Star imports (e.g. import java.util.*) are NOT ALLOWED -->
+        <module name="AvoidStarImport"/>
+        <!-- Redundant import statements are NOT ALLOWED -->
+        <module name="RedundantImport"/>
+        <!-- Unused import statements are NOT ALLOWED -->
+        <module name="UnusedImports"/>
+
+        <!-- ##### Javadocs requirements ##### -->
+        <!-- Requirements for Javadocs for classes/interfaces -->
+        <module name="JavadocType">
+            <!-- All public classes/interfaces MUST HAVE Javadocs -->
+            <property name="scope" value="public"/>
+            <!-- Add an exception for anonymous inner classes -->
+            <property name="excludeScope" value="anoninner"/>
+            <!-- Ignore errors related to unknown tags -->
+            <property name="allowUnknownTags" value="true"/>
+            <!-- Allow params tags to be optional -->
+            <property name="allowMissingParamTags" value="false"/>
+        </module>
+        <!-- Requirements for Javadocs for methods -->
+        <module name="JavadocMethod">
```

Comment:
These defaults for javadoc seem sensible to me. I am glad to see the flexibility around params/returns/throws.

Reply:
I'm a bit unhappy with making @param optional, but I can live with it.  Expect to see them added whenever I have to puzzle out a method's requirements.

Reply:
@mwoodiupui: I think `@param` should be highly encouraged, yes.  I was mostly hesitant about making `@param` required because I didn't want to slow this work down by forcing us to write a lot of code comments from scratch.  I think future PR(s) could add in any missing `@param` comments and then flip the switch to make them required.